### PR TITLE
docs: Added release notes and associated media for 0.38

### PR DESCRIPTION
**Pull request by garden-of-delete**
### SUMMARY
We would like to publish release notes with each major release. This is a first crack at sorting out the format for the release notes, starting with v0.38. These are the decision points that would benefit from feedback or objections, and will likely see iteration in future release notes:
- Multimedia embedded in the release notes vs no multimedia
- A single .md file with each new version's notes appended vs a separate .md file and directory for each release
- Directory structure for release notes
- How the release notes are linked to the top level README.md

The changes in this PR make a suggestion for each of those points.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N/A

### TEST PLAN
No testing required, no changes to codebase. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by benceorlai** [comment]
Thank Robert!

**Comment by mistercrunch** [comment]
See it rendered here:
https://github.com/garden-of-delete/incubator-superset/blob/master/RELEASING/release-notes-0-38/release-notes-0-38.md

Comments:
- file could be `/release-notes-0-38/README.md` so that github renders it
- expecting to see images embedded in the release notes but I don't see any (?)

**Comment by garden-of-delete** [comment]
> See it rendered here:
> https://github.com/garden-of-delete/incubator-superset/blob/master/RELEASING/release-notes-0-38/release-notes-0-38.md
> 
> Comments:
> 
> * file could be `/release-notes-0-38/README.md` so that github renders it when clicking on the folder
> * expecting to see images embedded in the release notes but I don't see any (?)

Addressed in most recent commits.

**Comment by benceorlai** [comment]
@garden-of-delete Github's MD supports  html tags, which gives you better control over the layout of images. Could you try something like:

File: RELEASING/release-notes-0-38/README.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+- fix(crud): set default extra value  (#[11262](https://github.com/apache/incubator-superset/pull/11262))
+- fix(sqla): allow 'unknown' type queries in explore view  (#[11365](https://github.com/apache/incubator-superset/pull/11365))
+- fix: prior npm font source had a glitch  (#[11724](https://github.com/apache/incubator-superset/pull/11724))
+
+
+## Breaking Changes
+[List of backwards incompatible changes](https://github.com/apache/incubator-superset/blob/master/UPDATING.md#0380)
+
+## Complete Changelog
+For the complete chnagelog please see [apache/incubator-superset/CHANGELOG.md](https://github.com/apache/incubator-superset/blob/master/CHANGELOG.md)
```
Comment (etr2460) [inline RELEASING/release-notes-0-38/README.md:319]: spelling nit: changelog

**Comment by mistercrunch** [comment]
Super nit-picky, but can we add borders to images? https://stackoverflow.com/questions/37349314/is-it-possible-to-add-border-to-image-in-github-markdown

Looks a bit rough here:
https://github.com/garden-of-delete/incubator-superset/tree/master/RELEASING/release-notes-0-38/

File: RELEASING/release-notes-0-38/README.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+
+Database CRUD screen refactor/redesign ([Roadmap](https://github.com/apache-superset/superset-roadmap/issues/14))
+- feat: add/edit database modal form sections UI  (#[10745](https://github.com/apache/incubator-superset/pull/10745))
+- feat: base tabbed modal for new database CRUD UI  (#[10668](https://github.com/apache/incubator-superset/pull/10668))
+
+
+Database list view refactor/redesign ([Roadmap](https://github.com/apache-superset/superset-roadmap/issues/55))
+- feat: database delete warning  (#[10800](https://github.com/apache/incubator-superset/pull/10800))
+<img src="media/10800.gif" width="50%"/>
```
Comment (ktmud) [inline RELEASING/release-notes-0-38/README.md:51]: Could you indent the image three spaces so it's wrapped under the list item?

Comment (garden-of-delete) [reply RELEASING/release-notes-0-38/README.md]: images are now wrapped under the bullet points.

**Review by mistercrunch — APPROVED** [review]

**Comment by mistercrunch** [comment]
Trying the old close/reopen trick since Cypress is not complying

**Comment by robdiciuccio** [comment]
Nit: would love to see `alt` tags on the images for improved accessibility.

**Review by robdiciuccio — APPROVED** [review]

**Pull request merged**